rdtools/normalization.py

```diff
@@ -0,0 +1,197 @@
+
+    Note: If T_cell and gamma_pdc are omitted, the temperature term will be
+          ignored.
+
+    Parameters
+    ----------
+    poa_global: Pandas Series (numeric)
+        Total effective plane of array irradiance.
+    P_stc: numeric
+        Module nameplate power at standard test condition.
+    T_cell: Pandas Series (numeric)
+        Measured or derived cell temperature [degrees celsius].
+        Time series assumed to be same frequency as poa_global.
+    G_stc: numeric, default value is 1000
+        Reference irradiance at standard test condition [W/m**2].
+    T_stc: numeric, default value is 25
+        Reference temperature at standard test condition [degrees celsius].
+    gamma_pdc: numeric, default is None
+        Linear array efficiency temperature coefficient [1 / degree celsius].
+
+    Returns
+    -------
+    dc_power: Pandas Series (numeric)
+        DC power determined by PVWatts v5 equation.
+    '''
+
+    dc_power = P_stc * poa_global / G_stc
+
+    if T_cell is not None and gamma_pdc is not None:
+        temperature_factor = 1 + gamma_pdc*(T_cell - T_stc)
+        dc_power = dc_power * temperature_factor
+
+    return dc_power
+
+
+def normalize_with_pvwatts(energy, pvwatts_kws):
+    '''
+    Normalize system AC energy output given measured poa_global and
+    meteorological data. This method uses the PVWatts V5 module model.
+
+    Energy timeseries and poa_global timeseries can be different granularities.
+
+    Parameters
+    ----------
+    energy: Pandas Series (numeric)
+        Energy time series to be normalized.
+    pvwatts_kws: dictionary
+        Dictionary of parameters used in the pvwatts_dc_power function.
+
+        PVWatts Parameters
+        ------------------
+        poa_global: Pandas Series (numeric)
+            Total effective plane of array irradiance.
+        P_stc: numeric
+            Module nameplate power at standard test condition.
+        T_cell: Pandas Series (numeric)
+            Measured or derived cell temperature [degrees celsius].
+            Time series assumed to be same frequency as poa_global.
+        G_stc: numeric, default value is 1000
+            Reference irradiance at standard test condition [W/m**2].
+        T_stc: numeric, default value is 25
+            Reference temperature at standard test condition [degrees celsius].
+        gamma_pdc: numeric, default is None
+            Linear array efficiency temperature coefficient [1 / degree celsius].
+
+    Returns
+    -------
+    normalized_energy: Pandas Series (numeric)
+        Energy divided by Sandia Model DC energy.
+    '''
+
+    if energy.index.freq is None:
+        freq = pd.infer_freq(energy.index)
+    else:
+        freq = energy.index.freq
+
+    dc_power = pvwatts_dc_power(**pvwatts_kws)
+
+    energy_dc = dc_power.resample(freq).sum()
+    normalized_energy = energy / energy_dc
+
+    return normalized_energy
+
+
+def sapm_dc_power(pvlib_pvsystem, irradiance):
```

Comment:
Makes sense. I will also include the required columns of met_data in the docstring.